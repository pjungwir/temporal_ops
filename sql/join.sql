-- Just a basic inner join

SELECT  a.id, b.id, a.valid_at * b.valid_at AS valid_at
FROM    a
JOIN    b
ON      a.id = b.id AND a.valid_at && b.valid_at
ORDER BY a.id, b.id, valid_at
