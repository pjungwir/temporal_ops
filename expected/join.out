-- Just a basic inner join
SELECT  a.id, b.id, a.valid_at * b.valid_at AS valid_at
FROM    a
JOIN    b
ON      a.id = b.id AND a.valid_at && b.valid_at
ORDER BY a.id, b.id, valid_at
 id | id | valid_at 
----+----+----------
  1 |  1 | [5,10)
  1 |  1 | [15,20)
  6 |  6 | [5,10)
  6 |  6 | [5,12)
  9 |  9 | [1,20)
(5 rows)

